Sources/COWMacros/COWExpansionFactory.swift

```diff
@@ -218,30 +223,196 @@ internal class COWExpansionFactory<Context: MacroExpansionContext> {
   
   // MARK: Utilities
   
-  private func createDerivedStorageTypeDecl() -> StructDeclSyntax {
-    let members = appliedStructValidVarDecls.map {
-      MemberBlockItemSyntax(decl: $0)
+  // If `appliedStructDecl` conforms to Equatable and manually implements `==`,
+  // the derived storage must also conform to Equatable (otherwise we hit a
+  // compiler bug that breaks the build).
+  // https://forums.swift.org/t/macro-multiple-matching-functions-xyz-error-though-the-expanded-code-has-no-duplicate-functions-and-compiles/68700/4
+  // FIXME: remove the workaround when the compiler bug is fixed.
+  private func applyEquatableWorkaroundForDerivedStorage(
+    members: inout [MemberBlockItemSyntax],
+    protocols: inout [InheritedTypeSyntax]
+  ) {
+    // Bail out if derived storage doesn't conform to Equatable.
+    guard protocols.contains(where: {
+      guard let name = $0.type.identifier else {
+        return false
+      }
+      return equatableProtocolNames.contains(name) ||
+             hashableProtocolNames.contains(name)
+    }) else {
+      return
     }
     
+    // Bail out if `appliedStructDecl` does not manually implement `==`.
+    guard let nestingEqualFunc = appliedStructDecl.memberBlock.members
```

Comment:
The process of getting `nestingEqualFunc` could be encapsulated into a function returning `FunctionDeclSyntax`, such that the ? in getting `nestingEqualFunc`'s `body` (line 277) could be eliminated.

Reply:
I believe that `as()` call in line 277 is unnecessary since I have already filtered non-`FunctionDeclSyntax` members by `.compactMap`. I have removed the redundant `as()`.

Reply:
As a side effect of replacing string literals when creating `==`, line 277 has been eliminated.